x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_pagination.ts

```diff
@@ -75,6 +75,19 @@ export function usePagination({ onPageChange, pageIndex, pageSize }: PaginationP
     [onChangePageIndex, pagination.pageIndex, pagination.pageSize]
   );
 
+  useEffect(() => {
```

Comment:
I did not want to refactor this component during BC, so I just added the logic to update the local state if the props changed. To be honest we should be able to remove the pagination state but I got scared because when i saw unit test failing.